client/my-sites/promote-post-i2/components/posts-list/header.tsx

```diff
@@ -5,46 +5,62 @@ type PostsListHeaderColumn = {
 	title: string | null;
 };
 
-export default function PostsListHeader() {
-	const columns: Array< PostsListHeaderColumn > = [
-		{
-			id: 'data',
-			title: translate( 'Post' ),
-		},
-		{
-			id: 'type',
-			title: translate( 'Type' ),
-		},
-		{
+type PostsListHeaderProps = {
+	postType: 'product' | 'post' | 'page';
+};
+
+export default function PostsListHeader( { postType: postType }: PostsListHeaderProps ) {
+	const getHeaderColumns = (): Array< PostsListHeaderColumn > => {
+		const columns = [
+			{
+				id: 'data',
+				title: translate( 'Post' ),
+			},
+			{
+				id: 'type',
+				title: translate( 'Type' ),
+			},
+		];
+
+		if ( postType === 'product' ) {
+			columns.push( {
+				id: 'sku',
+				title: translate( 'SKU' ),
+			} );
+			columns.push( {
+				id: 'price',
+				title: translate( 'Price' ),
+			} );
+		}
+
+		columns.push( {
 			id: 'publish-date',
 			title: translate( 'Publish date' ),
-		},
-		{
-			id: 'views',
+		} );
+
+		if ( postType === 'post' || postType === 'page' ) {
```

Comment:
Shouldn't likes and comments be the last element of the table? Here, the last element is visitors. 
We need to change this order to match the one in the table body.

And also, we are missing the view and promote columns. You can check the previous list.

Reply:
Fixed